package_locator/locator.py

```diff
@@ -12,6 +12,8 @@ def search_github_url_in_json_data(ecosystem, package, json_data):
             subdir = locate_subdir(ecosystem, package, repo_url)
             return repo_url, subdir
         except Exception as e:
+            if repo_url!= None:
+                return repo_url, None
```

Comment:
In `search_github_url_in_json_data`, we only return the repo_url if we could validate the repo by locating the package directory, otherwise we don't return anything.

The rationale here is that we can pass in any data to this function, e.g., package homepage, which may contain various GitHub URLs in the data for whatever reason. Therefore, a validation step here is a must. 

Let me know if I'm missing anything.